### DISCUSSION TITLE
Date Format Customization

### DISCUSSION THREAD

**Discussion by javawolfpack** [INGInious/INGInious#956] [Q&A] [answered]
Can the displayed date format be changed in the `frontend`? I just realized it is DD/MM/YYYY, which I am fine with, but being in the US will likely confuse most of our students.

**Comment by anthonygego** [comment, marked as answer]
Indeed, there remain some parts of the UI we don't always think about that cannot be localized, especially dates and timezones.  I've just renamed #526 to include the first one.

Ideally we should ensure to use a complete ISODate everywhere and use a JS library (or anything more appropriate) to print them in the preferred format. Python has made this thing quite annoying because it had not supported the full ISO 8601 date specs until version 3.11. Support was announced earlier but not with the same amount of digits, compromising tools interoperability.

As a quick workaround in the meantime, I would simply suggest to run a recursive `sed` on the INGInious folder to change the format:

    find <install_path> -type f -name "*.html" -exec sed -i "s/strftime(\([\"']\)%d\/%m/strftime(\1%m\/%d/g" {} +
    find <install_path> -type f -name "*.py" -exec sed -i "s/strftime(\([\"']\)%d\/%m/strftime(\1%m\/%d/g" {} +
    
This should be quite a safe operation.
